### ISSUE TITLE
sliver.dart': Failed assertion: line 291 pos 12: 'child != null': is not true.

### ISSUE THREAD

**Issue by rock-meister** [flutter/flutter#17824]
The error message requested that I report this exception.

I/flutter ( 9768): The following assertion was thrown during performLayout():
I/flutter ( 9768): 'package:flutter/src/widgets/sliver.dart': Failed assertion: line 291 pos 12: 'child != null': is
I/flutter ( 9768): not true.
I/flutter ( 9768):
I/flutter ( 9768): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter ( 9768): more information in this error message to help you determine and fix the underlying cause.
I/flutter ( 9768): In either case, please report this assertion by filing a bug on GitHub:
I/flutter ( 9768):   https://github.com/flutter/flutter/issues/new

PS> flutter doctor
Flutter (Channel beta, v0.3.2, on Microsoft Windows [Version 10.0.17134.48], locale en-US)

**Comment by zoechi** [comment]
There should be a new beta. Would you mind testing with the new version?
Can you provide code that allows to reproduce?

**Comment by rock-meister** [comment]
I am unable to reproduce with the new beta (Flutter 0.4.4 • channel beta).  Closing the issue until I can reliably reproduce it with the latest bits.

**Comment by chrisliebl** [comment]
I just ran into this issue with 0.4.4 on MacOS.

Flutter (Channel beta, v0.4.4, on Mac OS X 10.13.4 17E202, locale en-US)

I saw this error a few times, but the one that kept triggering it is using direction: Axis.vertical in a Flex. If I switch it to Axis.horizontal, there is a different error with a few suggestions. Following the suggestions and switching the child Expanded to a Flexible then brings back the "Failed assertion: line 291 pos 12" error.
